koku/masu/test/database/test_ocp_report_db_accessor.py

```diff
@@ -1032,3 +1034,52 @@ def test_populate_usage_costs_new_columns_no_report_period(self):
         with self.assertLogs("masu.database.ocp_report_db_accessor", level="INFO") as logger:
             self.accessor.populate_usage_costs("", "", start_date, end_date, self.provider_uuid)
             self.assertIn(expected, logger.output)
+
+    def test_populate_platform_and_worker_distributed_cost_sql_no_report_period(self):
+        """Test that updating monthly costs without a matching report period no longer throws an error"""
+        start_date = "2000-01-01"
+        end_date = "2000-02-01"
+        msg = "No report period for OCP provider, skipping platform_and_worker_distributed_cost_sql update."
+        context = {"provider_uuid": self.provider_uuid, "start_date": {start_date}}
+        expected = f"INFO:masu.database.ocp_report_db_accessor:{log_json(self.provider_uuid, msg, context)}"
+
+        with self.assertLogs("masu.database.ocp_report_db_accessor", level="INFO") as logger:
+            self.accessor.populate_platform_and_worker_distributed_cost_sql(
+                start_date, end_date, self.provider_uuid, {"platform_cost": True}
+            )
+            self.assertIn(expected, logger.output)
```

Comment:
I feel like this test is entirely _too_ specific for a unit test. It should be sufficient to just make sure `No report period for OCP provider, skipping platform_and_worker_distributed_cost_sql update` was logged. All this other stuff is not necessary.

It is fine to test that something was logged when that is the only outcome given the function input, but testing the actual logged content isn't great especially if we go back and change how things are logged.

Yes, I understand that there are other tests in this file that are similar to this one, but I would argue that those are not good tests either.

Reply:
Good point, I was just pushing a quick fix after updating to log_json. I just switched it to check if none.

Reply:
Yea thats good with me!

Reply:
Actually, thinking on this again, WE know that these params will result in the func returning `None` without doing anything, but this test does not assert that. This function returns `None` regardless. This is why I had mentioned to assert on just `No report period for OCP provider, skipping platform_and_worker_distributed_cost_sql update` because that is all we have to assert on given this path through the function.

The other reason I didn't want such a specific test assertion before was, we have a team goal to change how we log things and we still have not come up with some standard or guidance yet. So I don't see value in adding a test that checks every character of a logged message when it's possible we're gonna have to go back and re-write a test because it was overly specific.